datafusion/functions/src/datetime/date_part.rs

```diff
@@ -81,6 +82,9 @@ impl DatePartFunc {
                     Exact(vec![Utf8, Time32(Millisecond)]),
                     Exact(vec![Utf8, Time64(Microsecond)]),
                     Exact(vec![Utf8, Time64(Nanosecond)]),
+                    Exact(vec![Utf8, Interval(YearMonth)]),
+                    Exact(vec![Utf8, Interval(DayTime)]),
+                    Exact(vec![Utf8, Interval(MonthDayNano)]),
```

Comment:
@nrc can we add

```rs

                    Exact(vec![Utf8, Duration(Second)]),
                    Exact(vec![Utf8, Duration(Millisecond)]),
                    Exact(vec![Utf8, Duration(Microsecond)]),
                    Exact(vec![Utf8, Duration(Nanosecond)]),
                    Exact(vec![Utf8View, Duration(Second)]),
                    Exact(vec![Utf8View, Duration(Millisecond)]),
                    Exact(vec![Utf8View, Duration(Microsecond)]),
                    Exact(vec![Utf8View, Duration(Nanosecond)]),
```
here, then convert durations to Intervals — presumably `MonthDayNano` is easiest.

Reply:
These are added to this PR, Arrow PR coming soon

Reply:
Arrow PR: https://github.com/apache/arrow-rs/pull/6246